icoscp/station/station.py

```diff
@@ -737,10 +735,17 @@ def getList(theme=['AS', 'ES', 'OS'], ids=None):
         # Revert to default values, return all certified stations.
         theme = defaulttheme
 
+    icos_themes = [x for x in theme if x in defaulttheme]
+    non_icos_themes = [x for x in theme if x not in defaulttheme]
+
     # get station list, by default returns all icos stations
     stations = getIdList()
-    # filter by theme
-    stations = stations[stations.theme.isin(theme)]
+    # filter by theme and icosClass if applicable
+    stations_icos = stations[stations.theme.isin(icos_themes)]
+    stations_icos = stations_icos[stations_icos['icosClass'].notna() & stations_icos['icosClass'].isin(['1', '2'])]
+    stations_other = stations[stations.theme.isin(non_icos_themes)]
```

Comment:
Why stations_other is only then theme is in non_icos_themes? What about non-ICOS stations that are non-ICOS due to the absence of icosClass? What about ICOS stations with icosClass='Associated'? Should they not be included in the response?

Reply:
The `getList()` function returns either a list of stations according to the provided stations' ids or **a list of themed stations according to the provided themes.** 

The valid theme entries for `getList()` function can only be `['AS', 'ES', 'OS]`. Thus there shouldn't be any `non_icos_themes`. Moreover, the dataframes `stations_other` and `station_icos` shouldn't be concatenated and returned. We only need the `stations_icos`. (I don't believe we have any non_icos_themes anyway. The dataframe `stations_other` will be empty)

When you provide a list of **station ids** to the `getList()` function then you can get a station with icosClass='Associated' or icosClass=None. But, when you provide a list of themes you only get stations with icosClass='1' or icosClass='2'.

By default the function `getIdList()` is the one that returns all the stations regardless of icosClass.

Reply:
> Maybe @ukarst or @atverm know what icosClass ='Associated' means?

Reply:
> The `getList()` function returns either a list of stations according to the provided stations' ids or **a list of themed stations according to the provided themes.**
> 
> The valid theme entries for `getList()` function can only be `['AS', 'ES', 'OS]`. Thus there shouldn't be any `non_icos_themes`. Moreover, the dataframes `stations_other` and `station_icos` shouldn't be concatenated and returned. We only need the `stations_icos`. (I don't believe we have any non_icos_themes anyway. The dataframe `stations_other` will be empty)
> 
> When you provide a list of **station ids** to the `getList()` function then you can get a station with icosClass='Associated' or icosClass=None. But, when you provide a list of themes you only get stations with icosClass='1' or icosClass='2'.
> 
> By default the function `getIdList()` is the one that returns all the stations regardless of icosClass.

You are right @ZogopZ,  using the call `getIdList() ` will use `project = ICOS`, and then only the default themes will apply.

Reply:
'Associated' is another ICOS class but is currently only defined for ecosystem stations. So getList(['ES']) should also return those stations because they are ICOS stations.